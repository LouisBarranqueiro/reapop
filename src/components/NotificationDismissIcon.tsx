import React, {SyntheticEvent} from 'react'
import {Notification} from '../reducers/notifications/types'
import {Theme} from '../themes/types'
import {classnames} from '../constants'

type Props = {
    notification: Notification
    theme?: Theme
    dismissNotification: (event: SyntheticEvent) => void
}

const NotificationDismissIcon = (props: Props) => {
    const {theme, notification, dismissNotification} = props
    const styles = theme ? theme.notificationDismissIcon(notification) : {}
    return (
        <svg
            className={classnames.notificationDismissIcon}
            style={styles}
            onClick={dismissNotification}
            viewBox="4 4 8 8"
            fill="currentColor"
            data-testid="dismiss-icon"
        >
            <path
                fillRule="evenodd"
                d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
            />
        </svg>
    )
}

export default NotificationDismissIcon
