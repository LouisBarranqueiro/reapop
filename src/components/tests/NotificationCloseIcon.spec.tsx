import React from 'react'
import {act, render} from '@testing-library/react'

import atalhoTheme from '../../themes/atalho'
import {Notification, POSITIONS, STATUSES} from '../..'
import NotificationDismissIcon from '../NotificationDismissIcon'

describe('<NotificationDismissIcon/>', () => {
    const notification: Notification = {
        id: '1',
        position: POSITIONS.topLeft,
        status: STATUSES.none,
        buttons: [],
    }

    it.each([
        ['without theme', undefined],
        ['with theme', atalhoTheme],
    ])('should return dismiss icon %s', (_, theme) => {
        const {container} = render(
            <NotificationDismissIcon notification={notification} dismissNotification={jest.fn()} theme={theme} />
        )
        expect(container.innerHTML).toMatchSnapshot()
    })

    it('should call `dismissNotification` function when clicking on the icon', () => {
        const dismissNotifSpy = jest.fn()
        const {getByTestId} = render(
            <NotificationDismissIcon notification={notification} dismissNotification={dismissNotifSpy} />
        )
        const event = new MouseEvent('click', {bubbles: true})
        act(() => {
            getByTestId('dismiss-icon').dispatchEvent(event)
        })
        expect(dismissNotifSpy).toHaveBeenCalledTimes(1)
    })
})
