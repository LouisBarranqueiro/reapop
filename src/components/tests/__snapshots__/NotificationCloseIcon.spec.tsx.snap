// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationDismissIcon/> should return dismiss icon with theme 1`] = `"<svg class=\\"reapop__notification-dismiss-icon\\" style=\\"width: 12px; height: 12px; margin: 14px 10px; cursor: pointer; color: rgb(185, 194, 203); flex-shrink: 0;\\" viewBox=\\"4 4 8 8\\" fill=\\"currentColor\\" data-testid=\\"dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path></svg>"`;

exports[`<NotificationDismissIcon/> should return dismiss icon without theme 1`] = `"<svg class=\\"reapop__notification-dismiss-icon\\" viewBox=\\"4 4 8 8\\" fill=\\"currentColor\\" data-testid=\\"dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path></svg>"`;
